app/forms/state_file/nj_medical_expenses_form.rb

```diff
@@ -0,0 +1,15 @@
+module StateFile
+  class NjMedicalExpensesForm < QuestionsForm
+    set_attributes_for :intake,
+                       :medical_expenses
+
+    validates_numericality_of :medical_expenses, only_integer: true, message: :round_to_whole_number, if: -> { medical_expenses.present? }
+    validates :medical_expenses, presence: true, allow_blank: false, numericality: { greater_than_or_equal_to: 0 }, if: -> { medical_expenses.present? }
```

Comment:
[dust] I'm a little confused by the validating presence-true if present. It seems like nil is a valid value here, ie, they could move forward with an empty field. Is that correct?

If so, might be nice to have an explicit test to document that nil is a valid value

Reply:
The way I understand it - `if medical_expenses.present?` THEN `presence: true, allow_blank: false, greater_than_or_equal_to: 0`. Melanie can you confirm?

I tested something similar here: https://github.com/codeforamerica/vita-min/pull/4856/files#diff-62e5f3eca04b36aa7e3dd487d951844767312fbae9e56015100adc09a80f5a57, and Kevin from CFA recommended [shoulda-matchers](https://github.com/thoughtbot/shoulda-matchers) as a good way to test these.

Reply:
Currently this defaults to 0. This could be a good idea because of the requirement to show medical expenses on the review screen regardless of value and let the user go back and edit.

However, we can change it to allow blank, default to nil, and just show a 0 on the review screen if there is no value.

Maybe we should make some blanket decisions about defaults vs nil and add them to our internal eng practices doc?

Reply:
[dust] I am also confused why you have `validates :medical_expenses, presence: true ... if: -> { medical_expenses.present? }` - is this just "validate true if true"?

I think we have also been wondering about when to allow blank vs requiring a 0 and it's more challenging than I expected to make a blanket decision so we often handle it on a case by case basis :/